tests/testthat/test_prepareAnnotations.R

```diff
@@ -1,37 +1,150 @@
+    expect_equal(mcols(gr), mcols(expectedGR))
 })
 
 
-test_that("prepareAnnotations of genome library is a GRangesList", {
-    library(TxDb.Hsapiens.UCSC.hg38.knownGene)
-    gr <- prepareAnnotations(TxDb.Hsapiens.UCSC.hg38.knownGene)
+test_that("positive strand gives ascending exon_rank", {
+    gtf.file <- system.file("extdata", "Homo_sapiens.GRCh38.91_chr9_1_1000000.gtf", package = "bambu")
+  
+    gr <- prepareAnnotations(x = gtf.file)
+  
+    positive_check <- data.frame(txname = names(unlist(gr)), unlist(gr)) %>% 
+        filter(strand == "+") %>% 
+        group_by(txname) %>% 
+        summarise(validate = all(exon_rank == seq(n())))
+    
+    expect_true(all(positive_check$validate))
+})
 
-    expect_s4_class(gr, class = "CompressedGRangesList")
-    expect_named(mcols(gr), c("TXNAME", "GENEID", "eqClass"))
+
+test_that("positive strand gives descending exon_endRank", {
+    gtf.file <- system.file("extdata", "Homo_sapiens.GRCh38.91_chr9_1_1000000.gtf", package = "bambu")
+  
+    gr <- prepareAnnotations(x = gtf.file)
+  
+    positive_check <- data.frame(txname = names(unlist(gr)), unlist(gr)) %>% 
+        filter(strand == "+") %>% 
+        group_by(txname) %>% 
+        summarise(validate = all(exon_endRank == seq(n(),1)))
+  
+    expect_true(all(positive_check$validate))
 })
 
 
+test_that("negative strand gives descending exon_rank", {
+    gtf.file <- system.file("extdata", "Homo_sapiens.GRCh38.91_chr9_1_1000000.gtf", package = "bambu")
+  
+    gr <- prepareAnnotations(x = gtf.file)
+  
+    negative_check <- data.frame(txname = names(unlist(gr)), unlist(gr)) %>% 
+        filter(strand == "-") %>% 
+        group_by(txname) %>% 
+        summarise(validate = all(exon_rank == seq(n(),1)))
+  
+    expect_true(all(negative_check$validate))
+})
 
 
-test_that("prepareAnnotations from gtf file is GRangesList", {
+test_that("negative strand gives ascending exon_endRank", {
     gtf.file <- system.file("extdata", "Homo_sapiens.GRCh38.91_chr9_1_1000000.gtf", package = "bambu")
 
-    expectedGR <- readRDS(system.file("extdata", "annotationGranges_txdbGrch38_91_chr9_1_1000000.rds", package = "bambu"))
+    gr <- prepareAnnotations(x = gtf.file)
+  
+    negative_check <- data.frame(txname = names(unlist(gr)), unlist(gr)) %>% 
+        filter(strand == "-") %>% 
+        group_by(txname) %>% 
+        summarise(validate = all(exon_endRank == seq(n())))
+  
+    expect_true(all(negative_check$validate))
+})
 
+
+test_that("txid must be in EqClassById", {
+    gtf.file <- system.file("extdata", "Homo_sapiens.GRCh38.91_chr9_1_1000000.gtf", package = "bambu")
+  
     gr <- prepareAnnotations(x = gtf.file)
+    
+    check <- data.frame(mcols(gr)) %>% 
+        dplyr::select(txid, eqClassById) %>% 
+        rowwise() %>% 
+        mutate(validate = txid %in% eqClassById)
+    
+    expect_true(all(check$validate))
+})
+
+
+test_that("eqClassById is correct (tested for a few genes)", {
```

Comment:
can you add a comment describing what you test here?

Reply:
comment added.